### PR TITLE
Fix XCTest lifecycle in Xcode13.3

### PR DESCRIPTION
**Pull request by mimibuduo123**
The PR should summarize what was changed and why. Here are some questions to
help you if you're not sure:

 - What behavior was changed?
 - What code was refactored / updated to support this change?
 - What issues are related to this PR? Or why was this change introduced?

Checklist - While not every PR needs it, new features should consider this list:

 - [x] Does this have tests?
 - [ ] Does this have documentation?
 - [ ] Does this break the public API (Requires major version bump)?
 - [ ] Is this a new feature (Requires minor version bump)?

### CODE REVIEW

**Comment by mimibuduo123** [comment]
#1123

File: Sources/Quick/QuickTestObservation.swift [hunk 1]
```diff
@@ -12,16 +12,24 @@ import XCTest
 @objc internal final class QuickTestObservation: NSObject, XCTestObservation {
     @objc(sharedInstance)
     static let shared = QuickTestObservation()
+    
+    override init() {
+        super.init()
+        QuickSpec.enumerateSubclasses { specClass in
```
Comment (paweldudek) [inline Sources/Quick/QuickTestObservation.swift:18]: Note that the init will be called when types are instantiated (b/c of https://github.com/Quick/Quick/pull/1124/files#diff-6288283bc6481ff91171d7266acdfb23cfb0b8d5dba40176ba0e05046cd6b5f4R14). This results in the runtime subclass enumeration being called while classes are still being loaded into runtime. 

In our codebase this resulted in random crashes happening when the tests started running (I also tried a poor mans approach by wrapping this in an async call but that was already too late).

Comment (mimibuduo123) [reply Sources/Quick/QuickTestObservation.swift:18]: Could you tell me about the crash scene? A large number of case tests found no problems

Comment (mimibuduo123) [reply Sources/Quick/QuickTestObservation.swift:18]: and,   
```
    override init() {
        super.init()
        
        DispatchQueue.main.async {
            QuickSpec.enumerateSubclasses { specClass in
                // This relies on `_QuickSpecInternal`.
                (specClass as AnyClass).buildExamplesIfNeeded()
            }
        }
    }
```
is it better？

Comment (paweldudek) [reply Sources/Quick/QuickTestObservation.swift:18]: In our rather large code base (200k LoC with multiple modules) doing the init-based approach results in:

```
runtime assert: unexpected initialization order
```

This totally makes sense - this code is trying to walk through runtime while runtime is still being loaded (see screenshot below for details). 

Also as mentioned above: I tried wrapping this into an async call and that's too late - no tests are being registered 😞 

Screenshot 2022-03-18 at 13 14 54

**Pull request closed without merging**